### PR TITLE
new PolylineWalker(StepWalker) - [was #990]

### PR DESCRIPTION
**Pull request by th3w4y**
Refactoring in the context of the new walker/navigator concept

Fixes:
-  PolylineWalker class renamed to Polyline
-  new class PolylineWalker(StepWalker)
-  change few tests

### CODE REVIEW

**Comment by TheSavior** [comment]
:+1:

**Comment by th3w4y** [comment]
Additionally is plan to go below changes:

https://github.com/th3w4y/PokemonGo-Bot/pull/24

**Comment by th3w4y** [comment]
I will not merge https://github.com/th3w4y/PokemonGo-Bot/pull/24 to my dev in order to not be pulled in by this PR until I have some feedback on it...

File: pokemongo_bot/walkers/polyline_generator_tester.py [hunk 1]
```diff
@@ -3,10 +3,10 @@
...
 import haversine
 import polyline
-
-from polyline_walker import PolylineWalker
-
-a = PolylineWalker('Poststrasse+20,Zug,CH', 'Guggiweg+7,Zug,CH', 100)
+from math import ceil
+from polyline_generator import Polyline
+a = Polyline((47.1706378, 8.5167405), (47.1700271, 8.518072999999998), 100)
+print(a.points)
```
Comment (VirtualSatai) [inline pokemongo_bot/walkers/polyline_generator_tester.py]: Is this supposed to be here?

Comment (th3w4y) [inline pokemongo_bot/walkers/polyline_generator_tester.py]: This is not called by any code.. I left it for tests Inside there is a piece of code  that explains you how to check at https://developers.google.com/maps/documentation/utilities/polylineutility the points that were generated...

Comment (TheSavior) [inline pokemongo_bot/walkers/polyline_generator_tester.py]: It would be more beneficial to have this be an automated test that travis runs.

**Comment by douglascamata** [comment]
:+1:

**Pull request merged**